[BUG] NPE when running XQuery because "contextSequence" is null

**Describe the bug**

While refactoring function-documentation XQuery code I hit an NPE on exist 7.0.0-SNAPSHOT where exist 6.3.0 works flawlessly. 

With the affected version of function-documentation installed on a build from develop opening

http://localhost:8080/exist/apps/fundocs/view.html?uri=http://exist-db.org/xquery/xmldiff&location=java:org.exist.xquery.modules.xmldiff.XmlDiffModule#

Will result in 

```
HTTP ERROR 500 jakarta.servlet.ServletException: jakarta.servlet.ServletException: An error occurred while processing request to /exist/apps/fundocs/view.html: An unknown error occurred: Cannot invoke "org.exist.xquery.value.Sequence.getItemCount()" because "contextSequence" is null
```

Below is the relevant portion of the stacktrace

```
...
Caused by: java.lang.NullPointerException: Cannot invoke "org.exist.xquery.value.Sequence.getItemCount()" because "contextSequence" is null
	at org.exist.xquery.pragmas.Optimize.eval(Optimize.java:114)
	at org.exist.xquery.ExtensionExpression.eval(ExtensionExpression.java:75)
	at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:98)
	at org.exist.xquery.LetExpr.eval(LetExpr.java:99)
	at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:161)
	at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:289)
...
```

This issue blocks a release of exist 7. 

**Expected behavior**

_Valid_ XQuery 3.1 code that works on exist-db 6 to also work on exist-db 7

**To Reproduce**

If the above isn't working, please tell us the exact steps you took when you encountered the problem:
1. build exist-db v7 from current develop
2. start this instance
2. install the attached function-documentation xar
3. open http://localhost:8080/exist/apps/fundocs/view.html?uri=http://exist-db.org/xquery/xmldiff&location=java:org.exist.xquery.modules.xmldiff.XmlDiffModule#
4. See error

**Context (please always complete the following information)**

- Build: eXist-7.0.0-SNAPSHOT (633836a7c4243fb1188b8f40876a19f780992aa4)
- Java: 23.0.2 (Azul Systems, Inc.)
- OS: Mac OS X 15.3.2 (aarch64)

**Additional context**

- How is eXist-db installed? built from source
- Any custom changes in e.g. `conf.xml`? none

--- a/exist-core/src/main/java/org/exist/xquery/pragmas/Optimize.java
+++ b/exist-core/src/main/java/org/exist/xquery/pragmas/Optimize.java
@@ -90,7 +90,8 @@ public class Optimize extends AbstractPragma {
         boolean optimize = false;
         NodeSet originalContext = null;
 
-        if (contextSequence == null || contextSequence.isPersistentSet()) {    // don't try to optimize in-memory node sets!
+        // don't try to optimize in-memory node sets!
+        if (contextSequence == null || contextSequence.isPersistentSet()) {
             // contextSequence will be overwritten
             originalContext = contextSequence == null ? null : contextSequence.toNodeSet();
             if (cachedContext != null && cachedContext == originalContext) {
@@ -103,26 +104,24 @@ public class Optimize extends AbstractPragma {
             // in the current context
             if (useCached) {
                 optimize = cachedOptimize;
-            } else {
-                if (optimizables != null) {
-                    for (final Optimizable optimizable : optimizables) {
-                        final Sequence canBeOptimized = optimizable.canOptimizeSequence(contextSequence);
-                        if (canBeOptimized == null) {
-                            optimize = false;
-                            break;  // exit for-each loop
-                        }
-                        if (canBeOptimized.getItemCount() == contextSequence.getItemCount()) {
-                            // everything in sequence can be optimized
-                            optimize = true;  // so far so good, head to next for-loop of `optimizable`
-                        } else {
-                            // nothing or only some bits can be optimized
-                            optimize = false;
-                            break;  // exit for-each loop
-                        }
+            } else if (optimizables != null && contextSequence != null) {
+                for (final Optimizable optimizable : optimizables) {
+                    final Sequence canBeOptimized = optimizable.canOptimizeSequence(contextSequence);
+                    if (
+                            canBeOptimized == null ||
+                            canBeOptimized.getItemCount() != contextSequence.getItemCount()
+                    ) {
+                        optimize = false;
+                        break;  // exit for-each loop
                     }
+                    // so far so good,
+                    // everything in sequence can be optimized
+                    optimize = true;
+                    // head to next for-loop of `optimizable`
                 }
             }
         }
+
         if (optimize) {
             cachedContext = originalContext;
             cachedTimestamp = originalContext == null ? 0 : originalContext.getState();

